gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,280 @@
+ * limitations under the License.
+ */
+package com.google.gcloud.dns;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+
+import java.io.Serializable;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A class that represents Google Cloud DNS record set.
+ *
+ * <p>A unit of data that will be returned by the DNS servers.
+ *
+ * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets">Google Cloud DNS
+ * documentation</a>
+ */
+public class DnsRecord implements Serializable {
+
+  private static final long serialVersionUID = 2016011914302204L;
+  private final String name;
+  private final List<String> rrdatas;
+  private final Integer ttl;
+  private final DnsRecordType type;
+  private final String zoneName;
+  private final Long zoneId;
+
+  private DnsRecord() {
+    this.name = null;
+    this.rrdatas = null;
+    this.ttl = null;
+    this.type = null;
+    this.zoneName = null;
+    this.zoneId = null;
+  }
+
+  DnsRecord(Builder builder) {
+    this.name = builder.name;
+    this.rrdatas = ImmutableList.copyOf(builder.rrdatas);
+    this.ttl = builder.ttl;
+    this.type = builder.type;
+    this.zoneName = builder.zoneName;
+    this.zoneId = builder.zoneId;
+  }
+
+  /**
+   * Enum for the DNS record types supported by Cloud DNS.
+   *
+   * <p>Google Cloud DNS currently supports records of type A, AAAA, CNAME, MX NAPTR, NS, PTR, SOA,
+   * SPF, SRV, TXT.
+   *
+   * @see <a href="https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types">Cloud DNS
+   * supported record types</a>
+   */
+  public enum DnsRecordType {
+    A,
+    AAAA,
+    CNAME,
+    MX,
+    NAPTR,
+    NS,
+    PTR,
+    SOA,
+    SPF,
+    SRV,
+    TXT
+  }
+
+  public static class Builder {
+
+    private List<String> rrdatas = new LinkedList<>();
+    private String name;
+    private Integer ttl;
+    private DnsRecordType type;
+    private String zoneName;
+    private Long zoneId;
+
+    private Builder() {
+    }
+
+    /**
+     * Creates a builder and pre-populates attributes with the values from the provided DnsRecord
+     * instance.
+     */
+    public Builder(DnsRecord record) {
+      this.name = record.name;
+      this.ttl = record.ttl;
+      this.type = record.type;
+      this.zoneId = record.zoneId;
+      this.zoneName = record.zoneName;
+      this.rrdatas.addAll(record.rrdatas);
+    }
+
+    /**
+     * Adds a record to the record set. The records should be as defined in RFC 1035 (section 5) and
+     * RFC 1034 (section 3.6.1). Examples of records are available in Google DNS documentation.
+     *
+     * @see <a href="https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types">Google
+     * DNS documentation </a>.
+     */
+    public Builder add(String record) {
```

Comment:
Following the pattern that we use so far for similar cases (e.g. [`ProjectInfo` labels](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ProjectInfo.java#L200)) this should probably be:
`addRecord`, `removeRecord`, `clearRecords`, and `records(List<String>)`